src/ast/SemanticChecksVisitor.java

```diff
@@ -11,15 +11,16 @@
     private boolean isLegalForest;
     private boolean isLegalST;
     private boolean visitResult; //should be set to true at builder but any subsequent sets are to false
-    private Set<String> definitelyInitialized;
+    private Stack<Set<String>> definitelyInitialized;
```

Comment:
I'm not sure that I fully understand what you defined as initialized at what point, we should have a call about this